broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java

```diff
@@ -588,6 +603,27 @@ private RemotingCommand handlePutMessageResult(PutMessageResult putMessageResult
             return CompletableFuture.completedFuture(response);
         }
 
+        String[] topics = null;
+        int[] queueIds = null;
+        if (requestHeader.isMultiTopic()) {
+            topics = requestHeader.getTopic().split(MixAll.BATCH_TOPIC_SPLITTER); // decode topics
+            if (requestHeader.getQueueIds() == null) {
```

Comment:
For reference only. 
How about providing a common BATCH API style?
Encode/Decode the header and body in batch mode, and reuse the processor/storage layer.
It is better not to introduce the batch difficulty to the storage layer. 
For advanced feature batch consumequeue, it is not able to handle the inner message num, if the message has multi topic-queues.